kubetest/e2e.go

```diff
@@ -319,7 +319,7 @@ func run(deploy deployer, o options) error {
 		}))
 	}
 
-	if o.postTestCmd != "" {
+	if o.postTestCmd != "" && (o.test || (kubemarkUpErr == nil && o.testCmd != "")) {
```

Comment:
For true parity, preTestCmd looks like it's run twice?
- once if `--test` was passed: https://github.com/kubernetes/test-infra/blob/0bd5b532b2f3f1b98a2958a8bb059ab494471e5f/kubetest/e2e.go#L169-L176
- and again if  `--test-cmd` is passed (and kubemark didn't err if `--kubemark` was specified): https://github.com/kubernetes/test-infra/blob/0bd5b532b2f3f1b98a2958a8bb059ab494471e5f/kubetest/e2e.go#L226-L233